src/DotVVM.Framework/Hosting/DotvvmRequestContextExtensions.cs

```diff
@@ -28,6 +28,14 @@ public static string GetSpaContentPlaceHolderUniqueId(this IDotvvmRequestContext
         return DotvvmPresenter.DetermineSpaContentPlaceHolderUniqueId(context.HttpContext);
     }
 
+    /// <summary>
+    /// Gets cancellation token for the request
+    /// </summary>
+    public static CancellationToken GetCancellationToken(this IDotvvmRequestContext context)
+    {
+        var cancellationService = context.Services.GetService<IRequestCancellationTokenProvider>();
+        return cancellationService.GetCancellationToken(context);
```

Comment:
We should either call `GetRequiredService` or return `default(CancellationToken)` when the service is null. This may produce a NullReferenceException when the `IRequestCancellationTokenProvider` is missing

Reply:
Fixed by calling the `GetRequiredService`